### PR TITLE
add `yurtctl test init` cmd to setup OpenYurt cluster with kind

### LINKED ISSUE #754: [feature request] refactor of `yurtctl init` to support kind
**What would you like to be added**:
Enable `yurtctl init` to setup openyurt cluster through `kind` in purpose of e2e-tests.

**Why is this needed**:
As we discussed in #748

**others**
/kind feature

### PR DESCRIPTION
**Pull request by Congrool**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
As mentioned is #736, we will deprecate `yurtctl convert/revert` which is used by github actions. To run e2e tests in github actions, we still need a way to setup OpenYurt cluster with kind. It is this pr.

#### Which issue(s) this PR fixes:

Fixes #754 

#### Does this PR introduce a user-facing change?

Currently, the yurtctl test looks like as follows:
```release-note
$ _output/bin/yurtctl test -h
Tools for developers to test the OpenYurt Cluster

Usage:
  yurtctl test [flags]
  yurtctl test [command]

Available Commands:
  init        Using kind to setup OpenYurt cluster for test

Flags:
  -h, --help   help for test
```
```
$ _output/bin/yurtctl test init -h
Using kind to setup OpenYurt cluster for test

Usage:
  yurtctl test init [flags]

Flags:
      --cloud-nodes string          Comma separated list of cloud nodes. The control-plane will always be cloud node.If no cloud node specified, the control-plane node will be the only one cloud node.
      --cluster-name string         The cluster name of the new-created kind cluster. (default "openyurt")
  -h, --help                        help for init
      --ignore-error                Igore error when using openyurt version that is not officially released.
      --kind-config-path string     Specify the path where the kind config file will be generated. (default "/tmp/kindconfig.yaml")
      --kube-config string          Path where the kubeconfig file of new cluster will be stored. The default is ${HOME}/.kube/config.
      --kubernetes-version string   The version of kubernetes that the openyurt cluster is based on. (default "v1.21")
      --node-num int                Specify the node number of the kind cluster. (default 2)
      --openyurt-version string     The version of openyurt components. (default "latest")
      --use-local-images            If set, local images stored by docker will be used first.
```

So, that developers can setup openyurt cluster with cmd `yurtctl test init`.

CHANGLOG-2022.4.9
Revise the local_up_openyurt.sh to use `yurtctl test init` to setup OpenYurt cluster.

### CODE REVIEW

**Comment by Congrool** [comment]
This pr only enables `yurtctl init` to setup OpenYurt cluster with kind, and the github actions still use the `yurtctl convert`. I'll update it in another pr.

**Comment by rambohe-ch** [comment]
@Congrool Because usage of yurtctl init is changed, so how about to update user manual document of `yurtctl init` in the following link: https://openyurt.io/zh/docs/installation/yurtctl-init-join

**Comment by Congrool** [comment]
> @Congrool Because usage of yurtctl init is changed, so how about to update user manual document of `yurtctl init` in the following link: https://openyurt.io/zh/docs/installation/yurtctl-init-join

OK, I'll update it.

**Comment by Peeknut** [comment]
lgtm

But we want yurtctl init to just initialize the master node and yurtctl join to join the nodes. After the change, `yurtctl init kind` can be used to create a cluster, not just a master node. It seems complicate the use of yurtctl init. Does this go against the original idea? @rambohe-ch

**Comment by rambohe-ch** [comment]
> lgtm
> 
> But we want yurtctl init to just initialize the master node and yurtctl join to join the nodes. After the change, `yurtctl init kind` can be used to create a cluster, not just a master node. It seems complicate the use of yurtctl init. Does this go against the original idea? @rambohe-ch

@Peeknut It's a good question, and i agree with that `yurtctl init kind` create a cluster will go against the original idea. maybe we need to add another command for yurtctl to create a cluster. @Congrool @Peeknut Do you have any suggestions?

**Comment by Congrool** [comment]
🤔 How adout adding a new command called `yurtctl test`, and we can use `yurtctl test init` to create this kind cluster. In further, we can integrate e2e tests into it by `yurtctl test run`. @rambohe-ch @Peeknut

**Comment by Peeknut** [comment]
> 🤔 How adout adding a new command called `yurtctl test`, and we can use `yurtctl test init` to create this kind cluster. In further, we can integrate e2e tests into it by `yurtctl test run`. @rambohe-ch @Peeknut

agree

**Comment by rambohe-ch** [comment]
> 🤔 How adout adding a new command called `yurtctl test`, and we can use `yurtctl test init` to create this kind cluster. In further, we can integrate e2e tests into it by `yurtctl test run`. @rambohe-ch @Peeknut

@Congrool agree +1,  add a new command `yurtctl test`.

File: pkg/yurtctl/cmd/yurttest/kindinit/init.go [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+	}
+	return nil
+}
+
+func getGoBinPath() (string, error) {
+	gopath, err := exec.Command("bash", "-c", "go env GOPATH").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to get GOPATH, %s", err)
+	}
+	return strings.Join([]string{string(gopath), "bin"}, "/"), nil
```
Comment (rambohe-ch) [inline pkg/yurtctl/cmd/yurttest/kindinit/init.go:379]: return filepath.Join(gopath, "bin"), nil

**Comment by rambohe-ch** [comment]
@Congrool How about modify e2e test in github action to use `yurtctl test init`?

**Comment by Congrool** [comment]
Sure, I'll revise the github action.
/hold

**Review by rambohe-ch — APPROVED** [review]

**Pull request merged**